packages/rum/src/domain/record/serialize.ts

```diff
@@ -295,20 +433,45 @@ function serializeCDataNode(): CDataNode {
   }
 }
 
-function serializeChildNodes(node: Node, options: SerializeOptions): SerializedNodeWithId[] {
+export function serializeChildNodes(node: Node, options: SerializeOptions): SerializedNodeWithId[] {
+  const nodePrivacyLevel = options.parentNodePrivacyLevel
+    ? remapInternalPrivacyLevels(node, options.parentNodePrivacyLevel)
+    : getNodePrivacyLevel(node)
   const result: SerializedNodeWithId[] = []
-  forEach(node.childNodes, (childNode) => {
+  let shuffleElements = false
+
+  if (nodePrivacyLevel === NodePrivacyLevel.HIDDEN) {
+    return result
+  }
+
+  // To enhance privacy, we shuffle the child order for dropdowns. We don't want to shuffle text
+  // nodes around though, which should not exist alone within DATALIST/SELECT/OPTGROUP elements
+  if (nodePrivacyLevel === NodePrivacyLevel.MASK && node.nodeType === Node.ELEMENT_NODE) {
+    const tagName = (node as HTMLElement).tagName
+    if (tagName === 'DATALIST' || tagName === 'SELECT' || tagName === 'OPTGROUP') {
+      shuffleElements = true
+    }
+  }
```

Comment:
I'm a bit worried about having unstable serialzation. Let discuss about it further in the RFC.

Reply:
I agree there is a conflict here between unstable serialisation and privacy.

Reply:
Descoped from this PR for now